pkg/descheduler/node/node.go

```diff
@@ -121,3 +123,23 @@ func IsNodeUschedulable(node *v1.Node) bool {
 	}
 	return false
 }
+
+func PodFitAnyNode(pod *v1.Pod, nodes []*v1.Node) bool {
+	for _, node := range nodes {
+		nodeInfo := schedulercache.NewNodeInfo()
```

Comment:
Why do we want to update scheduler cache?

Reply:
I need a new variable of nodeInfo type to pass to `PodMatchNodeSelector`, so I used this. Is there a better way of doing this?

Reply:
I think, you are creating nodeInfo type here and in https://github.com/kubernetes-incubator/descheduler/pull/56/commits/790a9980854cdd9e51aca34bf0d7e36f12f16f06#diff-3b317ab7cb9a4335d438b4d3d0da8ac7R36 you are getting back the node from nodeInfo cache, instead you can pass the node directly.

Reply:
I copied that code for PodMatchNodeSelector from predicates package as it is, so I didn't change it. I have now modified it to take in node instead of nodeInfo, and now I do not need schedulercache anywhere.